nep-XX.mediawiki

```diff
@@ -0,0 +1,164 @@
+<pre>
+  NEP: XX
+  Title: Offline Message Signing and Verification Standard
+  Author: Adrian Fjellberg (atlazor), Alvaro Jara (ajara87)
+  Type: Standard
+  Status: Draft
+  Created: 2025-12-08
+</pre>
+
+==Abstract==
+
+This NEP specifies how the command `sign message` constructs and signs an offline message. It standardizes (1) the exact payload format (including an anti-transaction prefix/suffix and a random salt), (2) the use of the network magic as a domain separator via `GetSignData`, and (3) the minimum set of values that implementations expose so that signatures can be reproduced and verified by other tools.
+
+==Motivation==
+
+Neo users, node operators and external systems need to prove key ownership or authorize off-chain actions by signing messages instead of transactions. Without a standard, each implementation can invent its own payload and signing rules, making verification brittle and opening the door to cross-network replay.
+
+The goals of this NEP are:
+
+* Ensure that “message signatures” cannot be interpreted as valid transactions.
+* Bind signatures to a specific Neo network via the existing `networkMagic`.
+* Make it clear what fields a wallet or CLI should output (payload, keys, signatures, salt) so that other tools can verify them.
+
+==Specification==
+
+===Context and inputs===
+
+The standard applies to the CLI command:
+
+<pre>
+wallet sign message &lt;message&gt;
+</pre>
+
+A conforming implementation MUST:
```

Comment:
The standard is written in a way that ensures there is a single implementation. NEPs shouldn't specify CLI commands.